src/core/tests/Test_StatusHandler.py

```diff
@@ -509,64 +589,415 @@ def test_load_status_and_set_package_install_status(self):
+
+        substatus_file_data = substatus_file_data[0]["status"]["substatus"][0]
+        message_patch = json.loads(substatus_file_data["formattedMessage"]["message"])["patches"]
+        self.__assert_truncated_status_output(substatus_file_data, message_patch, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_WARNING, patch_count)
+        # Assert installation tomstone
+        # self.__assert_truncated_installation_tombstone(message_patches)
+        # Assert truncated error
+        self.__assert_truncated_error(substatus_file_data, error_count=0)
+
+    def test_installation_status_file_truncation_over_size_limit_low_priority_packages(self):
+        """ Test truncation logic will apply to installation over the size limit """
+        self.runtime.execution_config.operation = Constants.INSTALLATION
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+
+        patch_count = random.randint(780, 1100)
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.PENDING)
+        self.runtime.status_handler.set_installation_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_complete_status_output(substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_SUCCESS, patch_count)
+
+        # Assert installation truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+
+        substatus_file_data = substatus_file_data[0]["status"]["substatus"][0]
+        message_patch = json.loads(substatus_file_data["formattedMessage"]["message"])["patches"]
+        self.__assert_truncated_status_output(substatus_file_data, message_patch, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_WARNING, patch_count)
+        # Assert installation tombstone
+        # self.__assert_truncated_installation_tombstone(message_patches)
+        # Assert truncated error
+        self.__assert_truncated_error(substatus_file_data, error_count=0)
+
+    def test_installation_status_file_truncation_over_large_size_limit_with_extra_chars(self):
```

Comment:
Q: Looks like there was a similar test in Test_CoreMain, what is the difference between these 2 tests?

Reply:
this is a unit test, core main tests cover both assess and installation. done

Reply:
Comment thread is outdated as CoreMain tests have been squashed. Resolving to avoid noise.